main.go

```diff
@@ -76,8 +76,8 @@ func pipe(r io.Reader, w io.WriteCloser) {
 	fmt.Println("Closing pipe")
 }
 
-// pass a stopChan to the go routtine
-func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan chan struct{}) {
+// pass a stopChan to the go routine
+func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan *chan struct{}) {
```

Comment:
How does this change the behavior of this function? We aren't mutating the channel here, so what changes when we pass a pointer?